katdal/test/test_chunkstore.py

```diff
@@ -242,7 +242,7 @@ def test_chunk_zero_size(self):
     def test_put_chunk_noraise(self):
         name = self.array_name('x')
         self.store.create_array(name)
-        result = self.store.put_chunk_noraise(name, (1, 2), [])
+        result = self.store.put_chunk_noraise(name, (slice(1, 2),), np.ones(4))
```

Comment:
If I have this correctly, previously `BadChunk` was being returned by `put_chunk_noraise` because the slice had no explicit start and stop attributes with which to verify. However, now `BadChunk` is being returned because of the `dtype` mismatch. And the previous call should result in a `TypeError` being returned. Correct?

Reply:
Yes, except it's not a `dtype` mismatch, but a shape mismatch: `slice(1, 2)` => (1,) vs `np.ones(4)` => (4,). Put_chunk accepts the dtype of the given array (as long as it's not `object`).

This is an example of `BadChunk` abuse 😂  An invalid slice should never really be condoned, since it is a programmer error that must be fixed. It is even debatable if `BadChunk` should be ignored in the same way as `ChunkNotFound`. I just wanted a quick way to test `put_chunk_noraise`, which needs to raise a `ChunkStoreError` on demand.

I haven't really made a good case for `BadChunk`... It's basically something that should never happen :-)